index.js

```diff
@@ -78,6 +82,64 @@ class TrezorKeyring extends EventEmitter {
     return this.model;
   }
 
+  /**
+   * Gets the vendor, if known.
+   *
+   * @returns {"trezor" | "onekey"}
+   */
+  async getVendor() {
+    return this.vendor || (this.vendor = await this.fetchVendor());
+  }
+
+  /**
+   * fetch vendor by call getFeatures
+   * @param {object} features
+   * @returns {'onekey' | 'trezor'}
+   */
+  fetchVendor() {
+    return new Promise((resolve) => {
+      TrezorConnect.getFeatures()
+        .then((response) => {
+          if (!response.success) {
+            resolve(null);
+            return;
+          }
+          const vendor = this.__getVendor(response.payload);
+          resolve(vendor);
+        })
+        .catch(() => {
+          resolve(null);
+        });
+    });
+  }
+
+  /**
+   * get the vendor name of the hardware wallet
+   * @param {object} features
+   * @returns {'onekey' | 'trezor'}
+   */
+  __getVendor(features) {
```

Comment:
This is now a pure function depending only on `features` (nice!) and does not need to be an instance method anymore. 

So this can be broken out to an independent `const` and should be straightforward to add some  basic unit tests now as well :)

Reply:
With your guidance, `getVendor` function now looks very pure and I have added some unit test code 😊. 

Please continue reviewing the code

Reply:
Looking good now to me :+1:

Reply:
Do I need to wait for the MetaMask/devs review now ?

Can you help to push the progress? 

Thanks. 😄

Reply:
@originalix I hope I didn't get your hopes up too much - I'm was just doing this fly-by in position of a random person on the internet who'd like to help this get a proper consideration!